src/main/java/tc/oc/pgm/listeners/ChatDispatcher.java

```diff
@@ -102,6 +103,22 @@ public void sendAdmin(Match match, MatchPlayer sender, @Nullable @Text String me
       desc = "Send a direct message to a player",
       usage = "[player] [message]")
   public void sendDirect(Match match, MatchPlayer sender, Player receiver, @Text String message) {
+    MatchPlayer matchReceiver = manager.getPlayer(receiver);
+    if (matchReceiver != null) {
+      SettingValue option = matchReceiver.getSettings().getValue(SettingKey.PM);
+
+      switch (option) {
+        case PM_OFF:
+          if (!sender.getBukkit().hasPermission(Permissions.STAFF)) {
+            String name = receiver.getDisplayName(sender.getBukkit()) + ChatColor.RED;
+            Component component =
+                new PersonalizedTranslatable("command.message.blockedNoPermissions", name);
```

Comment:
I'm unsure what blockedNoPermissions message exactly is, but it shouldn't be a no perms, but rather a "player doesn't want DMS"

Reply:
It was already in the strings.

https://github.com/Electroid/PGM/blob/1a604f129d510489aa9ec0a398a31b3c787c139d/src/main/i18n/templates/strings.properties#L818